sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -1029,12 +1040,35 @@ public double getProgress() {
   private Progress getProgress() {
     synchronized (splitLock) {
       if (currentTracker instanceof RestrictionTracker.HasProgress) {
-        return ((HasProgress) currentTracker).getProgress();
+        return scaleProgress(
+            ((HasProgress) currentTracker).getProgress(),
+            currentWindowIterator.previousIndex(),
+            currentElement.getWindows().size());
```

Comment:
This should be the `stop index` since splitting limits the current window iterator so the progress will be wrong after a split happens.

Reply:
Added `windowStopIndex`.